### PR TITLE
feat: ingestion limits - add toggle feature

### PR DESCRIPTION
**Pull request by YounixM**
https://signoz-team.slack.com/archives/C02C7P6KE06/p1731427069496209?thread_ts=1731409354.155829&cid=C02C7P6KE06

----

> [!IMPORTANT]
> Add toggle feature for enabling/disabling daily and per-second ingestion limits with UI and data structure updates.
> 
>   - **Behavior**:
>     - Added toggle feature for daily and per-second limits in `MultiIngestionSettings.tsx` using `Switch` component.
>     - If no limits are set, the edit view closes without saving.
>   - **Data Handling**:
>     - Updated `LimitProps`, `AddLimitProps`, and `UpdateLimitProps` in `types.ts` to include `enabled` boolean for `day` and `second` limits.
>     - Modified `handleAddLimit` and `handleUpdateLimit` functions to handle the new toggle state.
>   - **UI Changes**:
>     - Updated styles in `IngestionSettings.styles.scss` to accommodate new toggle layout.
>     - Added visual indication for "NO LIMIT" state in the UI.
> 
> <sup>This description was created by </sup>[Ellipsis](https://www.ellipsis.dev?ref=SigNoz%2Fsignoz&utm_source=github&utm_medium=referral)<sup> for 6f09478f2451d500a325c319bd4fc97bd895960c. It will automatically update as commits are pushed.</sup>

### CODE REVIEW

**Review by grandwizard28 — APPROVED** [review]

File: frontend/src/container/IngestionSettings/MultiIngestionSettings.tsx [hunk 1]
```diff
@@ -684,50 +732,108 @@ function MultiIngestionSettings(): JSX.Element {
...
 																		<div className="daily-limit">
 																			<div className="heading">
-																				<div className="title"> Daily limit </div>
+																				<div className="title">
+																					Daily limit
+																					<div className="limit-enable-disable-toggle">
+																						<Form.Item name="enableDailyLimit">
+																							<Switch
+																								size="small"
+																								checked={activeSignal?.config?.day?.enabled}
```
Comment (ahmadshaheer) [inline frontend/src/container/IngestionSettings/MultiIngestionSettings.tsx:741]: nit: can we extract `activeSignal?.config?.day?.enabled` to a variable e.g. `isDailyLimitEnabled`

File: frontend/src/container/IngestionSettings/MultiIngestionSettings.tsx [hunk 2]
```diff
@@ -684,50 +732,108 @@ function MultiIngestionSettings(): JSX.Element {
...
 																		<div className="second-limit">
 																			<div className="heading">
-																				<div className="title"> Per Second limit </div>
+																				<div className="title">
+																					Per Second limit{' '}
+																					<div className="limit-enable-disable-toggle">
+																						<Form.Item name="enableSecondLimit">
+																							<Switch
+																								size="small"
+																								checked={activeSignal?.config?.second?.enabled}
```
Comment (ahmadshaheer) [inline frontend/src/container/IngestionSettings/MultiIngestionSettings.tsx:794]: nit: same here, can we extract `activeSignal?.config?.second?.enabled` to a variable e.g. `isPerSecondLimitEnabled`

**Review by ahmadshaheer — APPROVED** [review]
Reviewed the code. it looks good. just a minor comment. please feel free to make the changes in another PR if merging this PR is urgent. I wasn't able to check the UI locally, as the ingestion keys API throws "Unauthorized" error when I visit Ingestion settings.

File: frontend/src/container/IngestionSettings/MultiIngestionSettings.tsx (hunk 1 above)
Comment (YounixM) [reply frontend/src/container/IngestionSettings/MultiIngestionSettings.tsx:741]: Will take this up in the next PRs as we revamp the design @ahmadshaheer.

File: frontend/src/container/IngestionSettings/MultiIngestionSettings.tsx (hunk 2 above)
Comment (YounixM) [reply frontend/src/container/IngestionSettings/MultiIngestionSettings.tsx:794]: Will take this up in the next PRs as we revamp the design @ahmadshaheer

**Pull request merged**